pkg/controllers/binding/binding_controller.go

```diff
@@ -45,19 +46,23 @@ func (c *ResourceBindingController) Reconcile(ctx context.Context, req controlle
 
 	binding := &workv1alpha2.ResourceBinding{}
 	if err := c.Client.Get(context.TODO(), req.NamespacedName, binding); err != nil {
-		// The resource no longer exist, clean up derived Work objects.
+		// The resource no longer exist, in which case we stop processing.
 		if apierrors.IsNotFound(err) {
-			return helper.DeleteWorks(c.Client, labels.Set{
-				workv1alpha2.ResourceBindingNamespaceLabel: req.Namespace,
-				workv1alpha2.ResourceBindingNameLabel:      req.Name,
-			})
+			return controllerruntime.Result{}, nil
 		}
 
 		return controllerruntime.Result{Requeue: true}, err
 	}
 
 	if !binding.DeletionTimestamp.IsZero() {
-		return controllerruntime.Result{}, nil
+		if err := helper.DeleteWorks(c.Client, labels.Set{
+			workv1alpha1.ResourceBindingNamespaceLabel: req.Namespace,
+			workv1alpha1.ResourceBindingNameLabel:      req.Name,
```

Comment:
Those two label need to use `workv1alpha2`.

Reply:
+1
`workv1alpha1.` --> `workv1alpha2.`

Reply:
Done